src/views/studentcompleteregistration/studentcompleteregistration.jsx

```diff
@@ -0,0 +1,149 @@
+var api = require('../../lib/api');
+var intl = require('../../lib/intl.jsx');
+
+var Deck = require('../../components/deck/deck.jsx');
+var Progression = require('../../components/progression/progression.jsx');
+var Spinner = require('../../components/spinner/spinner.jsx');
+var Steps = require('../../components/registration/steps.jsx');
+
+require('./studentcompleteregistration.scss');
+
+var StudentCompleteRegistration = intl.injectIntl(React.createClass({
+    type: 'StudentCompleteRegistration',
+    getInitialState: function () {
+        return {
+            classroom: null,
+            formData: {},
+            registrationError: null,
+            step: 0,
+            waiting: false
+        };
+    },
+    advanceStep: function (formData) {
+        formData = formData || {};
+        this.setState({
+            step: this.state.step + 1,
+            formData: defaults({}, formData, this.state.formData)
+        });
+    },
+    componentDidUpdate: function (prevProps) {
+        if (prevProps.session.session !== this.props.session.session &&
+            this.props.session.session.permissions &&
+            this.props.session.session.permissions.student) {
+            var classroomId = this.props.session.session.user.classroomId;
+            api({
+                uri: '/classrooms/' + classroomId
+            }, function (err, body, res) {
+                if (err || res.statusCode === 404) {
+                    return this.setState({
+                        registrationError: this.props.intl.formatMessage({
+                            id: 'studentRegistration.classroomApiGeneralError'
+                        })
+                    });
+                }
+                this.setState({classroom: body});
+            }.bind(this));
+        }
+    },
+    register: function (formData) {
+        this.setState({waiting: true});
+        formData = defaults({}, formData || {}, this.state.formData);
+        var submittedData = {
+            birth_month: formData.user.birth.month,
+            birth_year: formData.user.birth.year,
+            gender: (
+                formData.user.gender === 'other' ?
+                formData.user.genderOther :
+                formData.user.gender
+            ),
+            country: formData.user.country,
+            is_robot: formData.user.isRobot
+        };
+        if (this.props.session.session.flags.must_reset_password) {
+            submittedData.password = formData.user.password;
+        }
+        api({
+            host: '',
+            uri: '/classes/student_update_registration/',
+            method: 'post',
+            useCsrf: true,
+            formData: submittedData
+        }, function (err, body) {
+            this.setState({waiting: false});
+            if (err) return this.setState({registrationError: err});
+            if (body.success) return this.advanceStep(formData);
+            this.setState({registrationError: (
+                <ul>
```

Comment:
For some reason, don't know why, it feels odd to have this html here, rather than being handled by `steps.RegistrationError`. From what I can tell, it's necessary to do the formatting here in order to get it to fit within the current paradigm, but I wonder if we should change how `RegistrationError` works in the future to accommodate one vs. multiple errors